Makefile

```diff
@@ -1,48 +1,50 @@
 .PHONY: dev lint complex coverage pre-commit yapf sort deploy destroy deps unit infra-tests integration e2e coverage-tests docs lint-docs build
+CWD := $(shell pwd)
 
-
-
+.ONESHELL:  # run all commands in a single shell, ensuring it runs within a local virtual env
 dev:
+	python3 -m venv .venv
+	. .venv/bin/activate
 	pip install --upgrade pip pre-commit poetry
 	pre-commit install
 	poetry config --local virtualenvs.in-project true
 	poetry install
+	npm ci
 
-lint:
+format:
```

Comment:
dont you need to add format to the top of the file under .PHONY?

Reply:
I'll add it. `.PHONY` instructs Make to always run the target `format` even if there's a file named `format` in the current directory. If there are no files named `format` or we don't build any file named `format`, there's no need for `.PHONY`.

It doesn't hurt adding it.